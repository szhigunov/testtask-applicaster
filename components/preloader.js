import React from 'react';
import {View, Text, ActivityIndicator, StyleSheet} from 'react-native';

export const Preloader = ({title}) => (
  <View style={styles.preloaderView}>
    <View>
      <ActivityIndicator />
      <Text style={styles.loadingMsg}>{title}</Text>
    </View>
  </View>
);

const styles = StyleSheet.create({
  preloaderView: {flex: 1, alignItems: 'center', justifyContent: 'center'},
  loadingMsg: {paddingTop: 10},
});
