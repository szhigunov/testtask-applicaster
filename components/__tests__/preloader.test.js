/**
 * @format
 */

import 'react-native';
import React from 'react';
import {Preloader} from '../preloader';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('Preloader renders correctly', () => {
  let root = renderer.create(<Preloader title="sometitle" />);
  expect(root.toJSON()).toMatchSnapshot();
});
