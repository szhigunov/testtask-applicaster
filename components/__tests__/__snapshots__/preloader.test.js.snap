// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preloader renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View>
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="small"
    />
    <Text
      style={
        Object {
          "paddingTop": 10,
        }
      }
    >
      sometitle
    </Text>
  </View>
</View>
`;
